File: little-bee/controller/router/menu.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_MENU = "/config/menu"
const URL_ROUTER_MENU_BY_ID = "/config/menu/:menu-id"
const URL_ROUTER_MENU_QUERY = "/config/menu/query"
const URL_ROUTER_MENU_ALL = "/config/menu/all"

func MenuRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_MENU, config.CreateMenu)
	r.GET(URL_ROUTER_MENU_BY_ID, config.RetrieveMenu)
	r.PUT(URL_ROUTER_MENU, config.UpdateMenu)
	r.DELETE(URL_ROUTER_MENU_BY_ID, config.DeleteMenu)
	r.PUT(URL_ROUTER_MENU_BY_ID, config.DeleteMenuWithFlag)
	r.POST(URL_ROUTER_MENU_QUERY, config.QueryMenu)
	r.GET(URL_ROUTER_MENU_ALL, config.GetAllMenus)
	return r
}

File: frontend/console/src/api/menu.js
import Axios from "@/http";

export function createMenu(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/menu', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveMenu(menuId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/menu/${ menuId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateMenu(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/menu', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteMenu(menuId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/menu/${ menuId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteMenuWithFlag(menuId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/menu/${ menuId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllMenus() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/menu/all').then(res => {
            resolve(res)
        })
    })
}

export function queryMenu (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/menu/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function batchCreateMenuAction (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/menu-action/batch', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function queryMenuAction (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/menu-action/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function assembledMenus() {
    return new Promise((resolve, reject) => {
        Axios.get('/user/assembled-menus').then(res => {
            resolve(res)
        })
    })
}

export function getUserAllActions() {
    return new Promise((resolve, reject) => {
        Axios.get('/user/all-actions').then(res => {
            resolve(res)
        })
    })
}